Add dynamic quant to the tutorial of PT2E quantization with X86Inductor (#2819)
pytorch/tutorials: prototype_source/pt2e_quant_x86_inductor.rst

The pytorch 2 export quantization flow uses the torch.export to capture the model into a graph and perform quantization transformations on top of the ATen graph.
This approach is expected to have significantly higher model coverage, better programmability, and a simplified UX.
TorchInductor is the new compiler backend that compiles the FX Graphs generated by TorchDynamo into optimized C++/Triton kernels.

> It supports both static and dynamic quantization. For the different between the two quantization types, please refer to the doc

Reviewer: ```suggestion
It supports both static and dynamic quantization. For the difference between the two quantization types, please refer to the doc
```

This approach is expected to have significantly higher model coverage, better programmability, and a simplified UX.
TorchInductor is the new compiler backend that compiles the FX Graphs generated by TorchDynamo into optimized C++/Triton kernels.

This flow of quantization 2 with Inductor supports both static and dynamic quantization. Static quantization works best for CNN models, like ResNet-50. And dynamic quantization is more suitable for NLP models, like RNN and BERT.

> For the difference between the two quantization types, please refer to the doc
> `here <https://pytorch.org/docs/stable/quantization.html#quantization-mode-support>`_.

Reviewer 2: ```suggestion
For the difference between the two quantization types, please refer to the `following page <https://pytorch.org/docs/stable/quantization.html#quantization-mode-support>`__.
```

  `multiplications are 7-bit x 8-bit <https://oneapi-src.github.io/oneDNN/dev_guide_int8_computations.html#inputs-of-mixed-type-u8-and-s8>`_. In other words, potential
  numeric saturation and accuracy issue may happen when running on CPU without Vector Neural Network Instruction.

The quantization config is for static quantization by default. To apply dynamic quantization, add an argument ``is_dynamic=True`` when getting the config.

> ::

Reviewer 2: please use code-block directives instead of ::. 
```suggestion
.. code-block:: python
```

Author: Thanks.
